gemonlinedb/include/gem/onlinedb/VFAT3ChannelConfiguration.h

```diff
@@ -0,0 +1,133 @@
+                     */
+                    bool armDacCorrectionPolarity : 1;
+
+                    /**
+                     * @brief Zero crossing comparator timing optimization:
+                     *        amplitude.
+                     */
+                    std::uint16_t zccDacCorrectionAmplitude : 6;
+
+                    /**
+                     * @brief Zero crossing comparator timing optimization:
+                     *        polarity.
+                     */
+                    bool zccDacCorrectionPolarity : 1;
+
+                    /**
+                     * @brief Mask the channel output for trigger and readout.
+                     */
+                    bool mask : 1;
+
+                    /**
+                     * @brief Enable the calibration pulse to the channel input.
+                     */
+                    bool cal : 1;
+                } fields;
+
+                /**
+                 * @brief Compares two register unions for equality.
+                 */
+                inline bool operator== (const ChannelRegisterValue other) const
+                {
+                    return registerValue == other.registerValue;
+                }
+            };
+
+            /* Check */
+            static_assert(sizeof(ChannelRegisterValue) == sizeof(std::uint16_t),
+                          "ChannelRegisterValue not packed in the expected way");
+
+        private:
+            std::array<ChannelRegisterValue, channelCount> registers;
+
+        public:
+            /**
+             * @brief Gets the configuration of the channel at the given index.
+             */
+            ChannelRegisterValue get(std::size_t channelIndex) const
+            {
+                return registers[channelIndex];
+            }
+
+            /**
+             * @brief Sets the configuration of the channel at the given index.
+             */
+            void set(std::size_t channelIndex, ChannelRegisterValue configuration)
+            {
+                registers[channelIndex] = configuration;
+            }
+
+            /**
+             * @brief Gets registers for serialization.
+             */
+            detail::RegisterData getRegisterData() const;
+
+            /**
+             * @brief Sets registers from serialized form.
+             */
+            void readRegisterData(const detail::RegisterData &data);
+
+            /**
+             * @brief Compares two VFAT3ChannelConfiguration for equality.
+             */
+            bool operator== (const VFAT3ChannelConfiguration &other) const;
+        };
+
+        template<>
+        class ConfigurationTraits<VFAT3ChannelConfiguration>
+        {
+        public:
+            static const char *extTableName() { return "GEM_VFAT3_CHANNEL_CONF"; }
+            static const char *typeName() { return "GEM VFAT3 Channel Conf Lookup Table"; }
+            static const char *kindOfPart() { return "GEM VFAT3"; }
```

Comment:
Is there a reason these are `char*` instead of `std::string`?
If a function needs them to be `char*` when passed, in, you can always `var.c_str()`